py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,339 @@
+            1 << 7: 'HSUPA',
+            1 << 8: 'HSPA',
+            1 << 9: 'HSPA+',  # 512
+            1 << 10: '1XRTT',
+            1 << 11: 'EVDO0',
+            1 << 12: 'EVDOA',
+            1 << 13: 'EVDOB',
+            1 << 14: 'LTE'  # 16384
+        }
+
+        # network registration states
+        # https://www.freedesktop.org/software/ModemManager/api/1.0.0/ModemManager-Flags-and-Enumerations.html#MMModem3gppRegistrationState
+        self.rstates = {
+            0: 'IDLE',
+            1: 'HOME',
+            2: 'SEARCHING',
+            3: 'DENIED',
+            4: 'UNKNOWN',
+            5: 'ROAMING'
+        }
+
+        self.bus = SystemBus()
+
+    def wwan_status_nm(self):
+        data = {}
+        # if there is no modem then disply STRING_NO_MODEM
+        full_text = self.py3.safe_format(
+            self.format_no_service, dict(status=STRING_NO_MODEM))
+        color = self.py3.COLOR_BAD
+
+        try:
+            modemmanager_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS)
+            modems = modemmanager_proxy.GetManagedObjects()
+        except:
+            full_text = self.py3.safe_format(self.format_no_service, data)
+            modems = {}
+
+        # browse modems objects
+        for objects in modems.items():
+            modem_path = objects[0]
+            modem_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, modem_path)
+
+            # we can maybe choose another selector
+            eqid = str(modem_proxy.EquipmentIdentifier)
+
+            # use selected modem or first find
+            if self.modem is None or self.modem == eqid:
+
+                # get status informations
+                try:
+                    status = modem_proxy.GetStatus()
+                except:
+                    # if no modem is selected, try all available
+                    continue
+
+                # start to build return data dict
+                try:
+                    data['signal'] = status['signal-quality'][0]
+                    self.py3.threshold_get_color(data['signal'], 'signal')
+                except:
+                    data['signal'] = STRING_UNKNOWN
+
+                try:
+                    highest_access_bit = 1 << (
+                        status['access-technologies'].bit_length() - 1)
+                    data['netgen'] = self.speed[highest_access_bit]
+                    self.py3.threshold_get_color(highest_access_bit, 'netgen')
+                except:
+                    data['netgen'] = STRING_UNKNOWN
+
+                # if registred on network, get operator name
+                try:
+                    if self.rstates[status['m3gpp-registration-state']] in (
+                            'HOME', 'ROAMING'):
+                        data['operator'] = status['m3gpp-operator-name']
+                    else:
+                        data['operator'] = STRING_UNKNOWN
+                except:
+                    data['operator'] = STRING_UNKNOWN
```

Comment:
This seems like something we don't have to do. We just return `{m3gpp_operator_name}` and not worrying about what state it is in. If we want to it to appear only in `HOME` or `ROAMING`. We could do this in the `format` later... Something like...
```
format = ''
format += '[\?if=m3gpp_operator_name=HOME {m3gpp_operator_name}]'
format += '[\?if=m3gpp_operator_name=ROAMING {m3gpp_operator_name}]'
format += '[\?if=m3gpp_operator_name=SEARCHING {m3gpp_operator_name}]'
etc... Users can pick when to print the name. :-) 
```